src/main/java/com/weather/service/WeatherService.java

```diff
@@ -19,19 +21,29 @@
 
     private final RestTemplate restTemplate;
 
-    public WeatherService(RestTemplate restTemplate) {
+    private final WeatherRepository weatherRepository;
+
+    private final WeatherForecastRepository weatherForecastRepository;
+
+    public WeatherService(RestTemplate restTemplate, WeatherRepository weatherRepository, WeatherForecastRepository weatherForecastRepository) {
         this.restTemplate = restTemplate;
+        this.weatherRepository = weatherRepository;
+        this.weatherForecastRepository = weatherForecastRepository;
     }
 
     public Weather getWeather(String city,String country)
     {
         URI url=new UriTemplate(ApplicationConstants.WEATHER_URL).expand(city,country,ApplicationConstants.API_KEY);
+        Weather weatherObj=invoke(url,Weather.class);
+        weatherRepository.save(weatherObj);
         return invoke(url,Weather.class);
     }
 
     public WeatherForecast getWeatherForecast(String city,String country) {
         URI url = new UriTemplate(FORECAST_URL).expand(city, country, ApplicationConstants.API_KEY);
-        return invoke(url, WeatherForecast.class);
+        WeatherForecast weatherForecastObj=invoke(url,WeatherForecast.class);
+        weatherForecastRepository.save(weatherForecastObj);
+        return weatherForecastObj;
     }
 
     private <T> T invoke(URI url, Class<T> responseType) {
```

Comment:
Happy to see your generic method. But RestTemplate class is already providing the generic methods by default. RestTemplate provide various method for respective of purpose of usage. 
So seems like you have created a generic method over generic method.

Reply:
Yes I learnt about of getForEntity() method of RestTemplate and i have implemented it.